### PR TITLE
Fix created events from scraper

### LINKED ISSUE #72: [Bug]: Fix scraper not parsing event duration
### Descriription

The Meetup scraper is not saving the `duration`, `url`, or `group` fields

### Expected behavior

All Event model fields have values from the scraper

### Steps to reproduce the behavior

Run the celery scraper task

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by joeriddles**
## Pull Request

**Description:**
- Fix scraper not parsing event duration
- Add custom template filter to render event duration from a `timedelta` object
- Fix Meetup scraper not saving group to events

**Related Issues:**
- fixes #72

**Checklist:**
- [x] All tests pass.
- [x] Code follows the project's coding standards.
- [x] Documentation has been updated.

### CODE REVIEW

**Comment by KGB33** [comment]
Is Postgres used in production? I ran into a migration issue when trying my Daggerized dev envrionment on this PR.

```
● start python manage.py runserver 0.0.0.0:8000 1.3s
    ✘ exec python manage.py migrate 1.3s
    ┃     super()._alter_field(                                                                                                                             
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 1134, in _alter_field                                        
    ┃     self.execute(                                                                                                                                     
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/postgresql/schema.py", line 48, in execute                                         
    ┃     return super().execute(sql, None)                                                                                                                 
    ┃            ^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                 
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 201, in execute                                              
    ┃     cursor.execute(sql, params)                                                                                                                       
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 122, in execute                                                    
    ┃     return super().execute(sql, params)                                                                                                               
    ┃            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                               
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in execute                                                     
    ┃     return self._execute_with_wrappers(                                                                                                               
    ┃            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                               
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers                                      
    ┃     return executor(sql, params, many, context)                                                                                                       
    ┃            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                       
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 100, in _execute                                                   
    ┃     with self.db.wrap_database_errors:                                                                                                                
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__                                                             
    ┃     raise dj_exc_value.with_traceback(traceback) from exc_value                                                                                       
    ┃   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 103, in _execute                                                   
    ┃     return self.cursor.execute(sql)                                                                                                                   
    ┃            ^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                   
    ┃   File "/usr/local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute                                                              
    ┃     raise ex.with_traceback(None)                                                                                                                     
    ┃ django.db.utils.ProgrammingError: cannot cast type integer to interval                                                                                
    ┃ LINE 1: ...R COLUMN "duration" TYPE interval USING "duration"::interval                                                                               
    ┃                                                              ^            
```

To reproduce run 
```
dagger -m github.com/SpokaneTech/SpokaneTech_Py@add-dagger call --src src/ --req requirements.txt dev --run --fresh-database as-service up
```

In this (`fix-event-duration`) branch. 

It'll spin up redis, Celery, and Postgres in the background for you.

**Comment by joeriddles** [comment]
> Is Postgres used in production? I ran into a migration issue when trying my Daggerized dev envrionment on this PR.

@KGB33 yes Postgres is used in prod. DurationField says it is supported in Postgres via the `interval` data type: https://docs.djangoproject.com/en/5.0/ref/models/fields/#durationfield.

What version of Postgres are you using in your Dagger dev environment?

**Comment by KGB33** [comment]
Its running `postgres:16`.

 I think the issue is that the migration script fails to convert the `duration` field from an Integer to an Interval.

**Comment by joeriddles** [comment]
> Its running `postgres:16`.
> 
> I think the issue is that the migration script fails to convert the `duration` field from an `integer` to an `interval`.

Ah that makes sense. I should probably switch to using postgres locally instead of sqlite to mirror prod. I will look into this, thank you!

**Comment by joeriddles** [comment]
@KGB33 I updated the migration to drop the IntegerField and re-add a DurationField. Normally this would be an issue due to the lost data but we're so early in this project that it's fine. Another way to fix it would be a custom migration but, again, I'm not worried about the lost data.

In a follow-up PR I'll plan on running [squashmigrations](https://docs.djangoproject.com/en/5.0/ref/django-admin/#django-admin-squashmigrations).

File: src/web/templates/web/event_detail.html [hunk 1]
```diff
@@ -15,7 +16,8 @@ <h1 class="mb-3 text-primary">
   </h1>
   
   <div class="mb-3 d-inline-block">
-    {% include 'spokanetech/partials/human_readable_datetime.htm' with object=object only %} for {{ object.duration }} hour{{ object.duration|pluralize }}
+    {% include 'spokanetech/partials/human_readable_datetime.htm' with object=object only %}
+    {% if object.duration %}for {{ object.duration|timedelta }}{% endif %}
   </div>
```
Comment (KGB33) [inline src/web/templates/web/event_detail.html:21]: I don't think this gets used when loading the event using Htmx. It has a different surrounding class, and is formatted weird. 
image
image


Navigating to the event using `http://localhost:8000/events/1` works & has the nice formatting, but its replaced when using the sidebar to view events.

**Comment by KGB33** [comment]
> @KGB33 I updated the migration to drop the IntegerField and re-add a DurationField. Normally this would be an issue due to the lost data but we're so early in this project that it's fine. Another way to fix it would be a custom migration but, again, I'm not worried about the lost data.
> 
> In a follow-up PR I'll plan on running [squashmigrations](https://docs.djangoproject.com/en/5.0/ref/django-admin/#django-admin-squashmigrations).

Sounds good! Thanks for fixing that for me.

File: src/web/templates/web/event_detail.html (hunk 1 above)
Comment (joeriddles) [reply src/web/templates/web/event_detail.html]: Thanks for pointing this out! I was only testing with the absolute URL. Fixed in this commit: https://github.com/SpokaneTech/SpokaneTech_Py/pull/71/commits/32a63d5cdc5d20cceab74caeb5d0909b2990e9c8

**Review by KGB33 — APPROVED** [review]

**Pull request merged**